areader/reader.go

```diff
@@ -1052,6 +1052,8 @@ func (ar *Reader) MergeArtifactDepends() (map[string]interface{}, error) {
 		deps, err := upd.GetUpdateDepends()
 		if err != nil {
 			return nil, err
+		} else if deps == nil {
```

Comment:
Question: I think in general that a nil error and type should not be returned from a `Get`, would it not be better to handle this lower in the call-stack?

Reply:
This isn't an error, the type-info Provides/Depends fields aren't required by the v3 specification.

Reply:
Good point. I think originally it was like this to represent the lack of this information from artifact format v2, but there's no reason it can't be empty map.

Reply:
I guess it's more about convenience than correct-ness. I can see that it's more convenient if it returns a map always, since in this context an empty map carries the same meaning as a nil pointer, but I don't really have a strong opinion which should be preferred.

Reply:
> I guess it's more about convenience than correct-ness. I can see that it's more convenient if it returns a map always, since in this context an empty map carries the same meaning as a nil pointer, but I don't really have a strong opinion which should be preferred.

Yup, my thoughts exactly. But no strong opinions here.